peer.go

```diff
@@ -1174,13 +1215,39 @@ func (p *peer) isActiveChannel(chanID lnwire.ChannelID) bool {
 	return ok
 }
 
+// storeError stores an error in our peer's buffer of recent errors with the
+// current timestamp. Errors are only stored if we have at least one active
+// channel with the peer to mitigate dos attack vectors where a peer costlessly
+// connects to us and spams us with errors.
+func (p *peer) storeError(err error) {
+	p.activeChanMtx.RLock()
+	channelCount := len(p.activeChannels)
```

Comment:
The current `peer.go` is a little wonky in that pending channels are also in `activeChannels` on startup due to `loadActiveChannels`. I have a slight refactor of this so that pending channels have `nil` in the `activeChannels` map and open channels actually have the channel data.  This could wait until that lands, or you could loop through all channels and check if one isn't pending.

Reply:
Link to PR? If that change is far off I'll update this, otherwise will wait for it to land.

Reply:
It will get into 0.10 but I don't see a reason to block this PR, I can just make the change on my end

Reply:
Cool, thanks! Tag me in the PR when it's up so I can specifically test this?

Reply:
Will do